java/src/main/java/ai/rapids/cudf/ParquetColumnWriterOptions.java

```diff
@@ -0,0 +1,406 @@
+
+  public static class ParquetStructColumnWriterOptions extends ParquetColumnWriterOptions {
+    protected ParquetStructColumnWriterOptions(AbstractStructBuilder builder) {
+      super(builder);
+    }
+
+  }
+
+  public static class ParquetListColumnWriterOptions extends ParquetColumnWriterOptions {
+    protected ParquetListColumnWriterOptions(ListBuilder builder) {
+      super(builder);
+    }
+  }
+
+  static abstract class NestedBuilder<T extends NestedBuilder> {
+    protected List<ParquetColumnWriterOptions> children = new ArrayList<>();
+    protected boolean isNullable = true;
+    protected String name = "";
+
+    /**
+     * Builder specific to build a Struct meta
+     * @param name
+     */
+    public NestedBuilder(String name, boolean isNullable) {
+      this.name = name;
+      this.isNullable = isNullable;
+    }
+
+    protected NestedBuilder() {}
+
+    protected ParquetColumnWriterOptions withColumn(String name, boolean isNullable) {
+      return new ParquetColumnWriterOptions(name, isNullable);
+    }
+
+    protected ParquetColumnWriterOptions withDecimal(String name, int precision,
+                                                     boolean isNullable) {
+      return new ParquetColumnWriterOptions(name, false, precision, isNullable);
+    }
+
+    protected ParquetColumnWriterOptions withTimestamp(String name, boolean isInt96,
+                                                       boolean isNullable) {
+      return new ParquetColumnWriterOptions(name, isInt96, 0, isNullable);
+    }
+
+    /**
+     * Set the list column meta.
+     * Lists should have only one child in ColumnVector, but the metadata expects a
+     * LIST column to have two children and the first child to be the
+     * {@link ParquetColumnWriterOptions#DUMMY_CHILD}.
+     * This is the current behavior in cudf and will change in future
+     * @param child
+     * @return this for chaining.
+     */
+    public T withListColumn(ParquetListColumnWriterOptions child) {
+      assert (child.getChildColumnOptions().length == 2) : "Lists can only have two children";
+      if (child.getChildColumnOptions()[0] != DUMMY_CHILD) {
+        throw new IllegalArgumentException("First child in the list has to be DUMMY_CHILD");
+      }
+      if (child.getChildColumnOptions()[1].getColumName().isEmpty()) {
+        throw new IllegalArgumentException("Column name can't be empty");
+      }
+      children.add(child);
+      return (T) this;
+    }
+
+    /**
+     * Set a child struct meta data
+     * @param child
+     * @return this for chaining.
+     */
+    public T withStructColumn(ParquetStructColumnWriterOptions child) {
+      for (ParquetColumnWriterOptions opt: child.getChildColumnOptions()) {
+        if (opt.getColumName().isEmpty()) {
+          throw new IllegalArgumentException("Column name can't be empty");
+        }
+      }
+      children.add(child);
+      return (T) this;
+    }
+
+    /**
+     * Set column name
+     * @param name
```

Comment:
Unless we describe what the name is marking it as a param does nothing. This applies to all of the other java doc comments for params that are empty.